src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateHistory.push({ ...addProperties(clone, action.extraData) });
```

Comment:
[CODE STYLE] - You don't need to spread the result of the 'addProperties' function when pushing to 'stateHistory'. The function already returns a new object, so spreading it into another object is redundant.